src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -691,7 +694,9 @@ public static bool TryReadBlock(ReadOnlySpan<byte> buffer, Stream furtherReads,
                     Span<byte> collatedHeader = dynamicHeaderSize <= StackAllocationThreshold ? stackalloc byte[StackAllocationThreshold].Slice(0, dynamicHeaderSize) : arrayPoolBuffer.AsSpan(0, dynamicHeaderSize);
 
                     buffer[FieldLocations.DynamicData..].CopyTo(collatedHeader);
-                    int realBytesRead = furtherReads.Read(collatedHeader[remainingBufferLength..]);
+
+                    Debug.Assert(bytesToRead == collatedHeader[remainingBufferLength..].Length);
+                    int realBytesRead = furtherReads.ReadAtLeast(collatedHeader[remainingBufferLength..], bytesToRead, throwOnEndOfStream: false);
 
                     if (realBytesRead != bytesToRead)
```

Comment:
I'm wondering about this if check.

Now that we are using ReadAtLeast, `realBytesRead` should be guaranteed to be the exact same number as `bytesToRead` if the stream still had enough bytes to read. This means that in the case where the number of bytes left to read in the stream is smaller than bytesToRead, we will return false here. But what will happen with those bytes we read? Aren't we losing information?

Reply:
We shouldn't lose information. In this situation, we'd be dealing with a CD entry at the end of a stream, with a total size which extends beyond the end of the stream. That wouldn't be a valid entry, so TryReadBlock would return false. We'd lose the bytes read into `furtherReads`, but that shouldn't be a problem - we've already decided not to process them.

Reply:
Got it. Thanks.